src/main.rs

```diff
@@ -305,68 +300,78 @@ async fn handle_web_req(
 
             let mut lnd = handler.client;
 
-            let headers = Headers(req.headers().to_owned());
-            let query = {
-                let uri = req.uri();
-                if let Some(query) = uri.query() {
-                    Some(&query.to_owned());
-                }
-                None
-            };
-            let body = req.into_body();
-            let bytes = hyper::body::to_bytes(body).await?;
-            dbg!(&bytes); // this is correct by my accounts
-            let reader = &*bytes;
-            let proposal = UncheckedProposal::from_request(reader, query, headers).unwrap();
+            // extract original PSBT from HTTP request
+            // TODO: `UncheckedProposal` is not really the proposal, but the original PSBT? Is the naming wrong?
+            // TODO: Unneeded and/or undocumented traits in bip-78 library? (`Proposal`, `Headers`)
+            // TODO: We should really save the original PSBT just in case the sender does not sign
```

Comment:
Pretty much. This is most important in the case when the receiver is a payment processor that generates bip21 URIs on request, called a non-interactive receiver. If you're scanning someone next to you's phone it's not so important to broadcast the `UncheckedProposal.extract_tx()` in the failure case or even make sure it can be broadcast because humans will be able to interact to solve the fail case by trying again.

Reply:
As previously mentioned, if `UncheckedProposal` is renamed to `OriginalPsbtRequest` or something alike, it would be less confusing.